### PR TITLE
object_storage,s3: fix head_bucket call according to docs

### PR DESCRIPTION
**Pull request by rominf**
# About this change - What it does

object_storage,s3: fix head_bucket call according to docs.

# Why this way

Previous method of error handling was wrong.

### CODE REVIEW

**Comment by rikonen** [comment]
That 403 handling is intentional change from 2015 (https://github.com/Aiven-Open/pghoard/commit/d6c4cdfb702ec94785798ebb02af49e2aa950df5) and I suspect something will break for us due to the way how we set up the bucket policies if this is merged.

File: rohmu/object_storage/s3.py [hunk 1]
```diff
@@ -542,12 +543,11 @@ def check_or_create_bucket(self) -> None:
...
             self.s3_client.head_bucket(Bucket=self.bucket_name)
         except botocore.exceptions.ClientError as ex:
+            # https://docs.aws.amazon.com/AmazonS3/latest/API/API_HeadBucket.html
             status_code = ex.response.get("ResponseMetadata", {}).get("HTTPStatusCode")
             if status_code == 301:
                 raise InvalidConfigurationError(f"Wrong region for bucket {self.bucket_name}, check configuration")
-            elif status_code == 403:
-                self.log.warning("Access denied on bucket check, assuming write permissions")
-            elif status_code == 404:
+            elif status_code in {HTTPStatus.BAD_REQUEST, HTTPStatus.FORBIDDEN, HTTPStatus.NOT_FOUND}:
```
Comment (giacomo-alzetta-aiven) [inline rohmu/object_storage/s3.py:550]: Just to keep track of the discussion: in some places we rely on the current behaviour for 403/Forbidden of not creating a bucket and simply logging a message, so need to double check how to fix this part without breaking other stuff.

**Comment by rominf** [comment]
It it good now?

**Review by giacomo-alzetta-aiven — APPROVED** [review]
LGTM

**Pull request merged**